examples/taproot-psbt.rs

```diff
@@ -0,0 +1,556 @@
+        let script = Self::timelock_script(locktime, beneficiary_key);
+        let leaf_hash = TapLeafHash::from_script(&script, LeafVersion::TapScript);
+
+        let taproot_spend_info = TaprootBuilder::new().add_leaf(0, script.clone())?.finalize(&self.secp, internal_keypair.public_key())?;
+        self.current_spend_info = Some(taproot_spend_info.clone());
+        let script_pubkey = Script::new_v1_p2tr(&self.secp, taproot_spend_info.internal_key(), taproot_spend_info.merkle_root());
+        let absolute_fees_in_sats = 1000;
+        let value = input_utxo.amount_in_sats - absolute_fees_in_sats;
+        
+        // Spend a normal BIP86-like output as an input in our inheritance funding transaction
+        let tx = generate_bip86_key_spend_tx(&self.secp, self.master_xpriv, input_utxo, vec![TxOut {
+            script_pubkey: script_pubkey.clone(),
+            value,
+        }])?;
+
+        // CREATOR + UPDATER
+        let next_tx = Transaction {
+            version: 2,
+            lock_time: 0,
+            input: vec![TxIn {
+                previous_output: OutPoint {
+                    txid: tx.txid(),
+                    vout: 0,
+                },
+                script_sig: Script::new(),
+                sequence: 0xFFFFFFFD, // enable locktime and opt-in RBF
+                witness: Witness::default(),
+            }],
+            output: vec![],
+        };
+        let mut next_psbt = Psbt::from_unsigned_tx(next_tx)?;
+        let mut origins = BTreeMap::new();
+        origins.insert(
+            beneficiary_key,
+            (
+                vec![leaf_hash],
+                (
+                    self.beneficiary_xpub.fingerprint(),
+                    derivation_path,
+                ),
+            ),
+        );
+        let ty = PsbtSighashType::from_str("SIGHASH_ALL")?;
+        let mut tap_scripts = BTreeMap::new();
+        tap_scripts.insert(taproot_spend_info.control_block(&(script.clone(), LeafVersion::TapScript)).unwrap(), (script, LeafVersion::TapScript));
+
+        let input = Input {
+            witness_utxo: {
+                let script_pubkey = script_pubkey;
+                let amount = Amount::from_sat(value);
+    
+                Some(TxOut {
+                    value: amount.to_sat(),
+                    script_pubkey,
+                })
+            },
+            tap_key_origins: origins,
+            tap_merkle_root: taproot_spend_info.merkle_root(),
+            sighash_type: Some(ty),
+            tap_internal_key: Some(internal_keypair.public_key()),
+            tap_scripts,
+            ..Default::default()
+        };
+
+        next_psbt.inputs = vec![input];
+        self.next_psbt = Some(next_psbt.clone());
+
+        self.next += 1;
+        Ok((tx, next_psbt))
+    }
+
+    fn refresh_tx(&self, locktime_delta: u32) -> Result<(Transaction, Psbt), Box<dyn std::error::Error>> {
```

Comment:
Is this still a todo?

Reply:
Yes, indeed! Sorry, just getting back to this today :)